src/accountsdb/snapshots.zig

```diff
@@ -853,22 +1463,33 @@ pub const StatusCache = struct {
     }
 };
 
+pub const CompressionMethod = enum {
+    zstd,
+
+    pub inline fn extension(method: CompressionMethod) []const u8 {
+        return switch (method) {
+            .zstd => "zst",
+        };
+    }
+};
+
 /// information on a full snapshot including the filename, slot, and hash
 ///
 /// Analogous to [SnapshotArchiveInfo](https://github.com/anza-xyz/agave/blob/59bf1809fe5115f0fad51e80cc0a19da1496e2e9/runtime/src/snapshot_archive_info.rs#L44)
 pub const FullSnapshotFileInfo = struct {
-    filename: []const u8,
     slot: Slot,
-    hash: []const u8,
-
-    const Self = @This();
+    hash: Hash,
+    comptime compression: CompressionMethod = .zstd,
 
-    pub fn deinit(self: *Self, allocator: std.mem.Allocator) void {
-        allocator.free(self.filename);
-    }
+    const FULL_SNAPSHOT_NAME_FMT = "snapshot-{[slot]d}-{[hash]s}.tar.{[extension]s}";
```

Comment:
What are these square brackets for, just visual labelling? Are they ignored by the formatter?

Reply:
The `{[name]}` syntax are a way of specifying named format arguments. If you look at `FULL_SNAPSHOT_NAME_MAX_LEN` just below, you'll see the format argument tuple is instead a record struct, where each argument is assigned to a name corresponding to a bracketed name in the format string.

Reply:
That's awesome, I didn't know we had that in zig.